plugins/tpm/fu-plugin-tpm.c

```diff
@@ -11,10 +11,43 @@
 
 #include "fu-tpm-device.h"
 
+struct FuPluginData {
+	gboolean		 has_tpm;
+	gboolean		 has_tpm_v20;
+};
+
 void
 fu_plugin_init (FuPlugin *plugin)
 {
+	fu_plugin_alloc_data (plugin, sizeof (FuPluginData));
 	fu_plugin_set_build_hash (plugin, FU_BUILD_HASH);
 	fu_plugin_add_udev_subsystem (plugin, "tpm");
 	fu_plugin_set_device_gtype (plugin, FU_TYPE_TPM_DEVICE);
+	fu_plugin_add_rule (plugin, FU_PLUGIN_RULE_REQUIRED_FOR, "hsi");
+}
+
+void
+fu_plugin_device_added (FuPlugin *plugin, FuDevice *dev)
+{
+	FuPluginData *data = fu_plugin_get_data (plugin);
+	data->has_tpm = TRUE;
+	if (g_strcmp0 (fu_tpm_device_get_family (FU_TPM_DEVICE (dev)), "2.0") == 0)
+		data->has_tpm_v20 = TRUE;
+}
+
+gboolean
+fu_plugin_add_security_attrs (FuPlugin *plugin, GPtrArray *attrs, GError **error)
+{
+	FuPluginData *data = fu_plugin_get_data (plugin);
+	FwupdSecurityAttr *attr = fwupd_security_attr_new ("org.trustedcomputinggroup.Tpm");
+	fwupd_security_attr_set_level (attr, FWUPD_SECURITY_ATTR_LEVEL_CRITICAL);
+	fwupd_security_attr_set_name (attr, "TPM");
+	if (data->has_tpm_v20) {
+		fwupd_security_attr_add_flag (attr, FWUPD_SECURITY_ATTR_FLAG_SUCCESS);
+		fwupd_security_attr_set_result (attr, "v2.0");
+	} else if (data->has_tpm) {
+		fwupd_security_attr_set_result (attr, "Not in v2.0 mode");
+	}
```

Comment:
Doesn't there need to be a 3rd case of doesn't have TPM?

Reply:
We could have `fwupd_security_attr_set_result(attr, "Not detected")` I guess, but at the moment it should just show `Failed`.

Reply:
Well that's probably enough, just wanted to make sure the test was going to show up, and since it's required for HSI I guess it will